processor/logcountprocessor/processor.go

```diff
@@ -0,0 +1,176 @@
+	logger    *zap.Logger
+	cancel    context.CancelFunc
+	mux       sync.Mutex
+}
+
+// newProcessor returns a new processor.
+func newProcessor(config *ProcessorConfig, consumer consumer.Logs, matchExpr *Expression, attrExprs map[string]*Expression, logger *zap.Logger) *processor {
+	return &processor{
+		config:    config,
+		matchExpr: matchExpr,
+		attrExprs: attrExprs,
+		counter:   NewLogCounter(),
+		consumer:  consumer,
+		logger:    logger,
+	}
+}
+
+// Start starts the processor.
+func (p *processor) Start(_ context.Context, _ component.Host) error {
+	ctx, cancel := context.WithCancel(context.Background())
+	p.cancel = cancel
+	go p.handleMetricInterval(ctx)
+
+	return nil
+}
+
+// Capabilities returns the consumer's capabilities.
+func (p *processor) Capabilities() consumer.Capabilities {
+	return consumer.Capabilities{MutatesData: false}
+}
+
+// Shutdown stops the processor.
+func (p *processor) Shutdown(_ context.Context) error {
+	p.cancel()
+	return nil
+}
+
+// ConsumeLogs processes the logs.
+func (p *processor) ConsumeLogs(ctx context.Context, pl plog.Logs) error {
+	p.mux.Lock()
+	defer p.mux.Unlock()
+
+	records := convertToRecords(pl)
+	for _, record := range records {
+		if p.matchRecord(record) {
+			resource := p.extractResource(record)
+			attrs := p.extractAttributes(record)
+			p.counter.Add(resource, attrs)
+		}
+	}
+
+	return p.consumer.ConsumeLogs(ctx, pl)
+}
+
+// handleMetricInterval sends metrics at the configured interval.
+func (p *processor) handleMetricInterval(ctx context.Context) {
+	ticker := time.NewTicker(p.config.Interval)
+	defer ticker.Stop()
+
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		case <-ticker.C:
+			p.sendMetrics(ctx)
+		}
+	}
+}
+
+// sendMetrics sends metrics to the consumer.
+func (p *processor) sendMetrics(ctx context.Context) {
+	p.mux.Lock()
+	defer p.mux.Unlock()
+
+	metrics := p.createMetrics()
+	p.counter.Reset()
+
+	if err := sendMetrics(ctx, p.config.ID(), metrics); err != nil {
+		p.logger.Error("Failed to send metrics", zap.Error(err))
+	}
+}
+
+// matchRecord returns true if the record matches the configured expression.
+func (p *processor) matchRecord(record Record) bool {
+	matches, err := p.matchExpr.Match(record)
+	if err != nil {
+		p.logger.Debug("Failed to evaluate match expression", zap.Error(err))
+		return false
+	}
+
+	return matches
+}
+
+// extractAttributes extracts attributes from the record.
+func (p *processor) extractAttributes(record Record) map[string]interface{} {
+	attrs := map[string]interface{}{}
+	for key, expression := range p.attrExprs {
+		value, err := expression.Evaluate(record)
+		if err != nil {
+			p.logger.Debug("Failed to evaluate attribute expression", zap.Error(err))
+			continue
+		}
+		attrs[key] = value
+	}
+	return attrs
+}
+
+// extractResource extracts the resource from the record.
+func (p *processor) extractResource(record Record) map[string]interface{} {
+	value, ok := record[resourceField].(map[string]interface{})
```

Comment:
The idea here is that resource is expected to be a map on the record. However, there's nothing to guarantee this key will exist or that everything will have a resource. I'm going to update this to return an empty map rather than nil.